oxd-server/src/main/java/org/gluu/oxd/server/op/RsModifyOperation.java

```diff
@@ -0,0 +1,157 @@
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class RsModifyOperation extends BaseOperation<RsModifyParams> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RsModifyOperation.class);
+
+    /**
+     * Constructor
+     *
+     * @param command command
+     */
+    protected RsModifyOperation(Command command, final Injector injector) {
+        super(command, injector, RsModifyParams.class);
+    }
+
+
+    @Override
+    public IOpResponse execute(final RsModifyParams params) throws Exception {
+        validate(params);
+
+        Rp site = getRp();
+
+        PatProvider patProvider = new PatProvider() {
+            @Override
+            public String getPatToken() {
+                return getUmaTokenService().getPat(params.getOxdId()).getToken();
+            }
+
+            @Override
+            public void clearPat() {
+                // do nothing
+            }
+        };
+
+        org.gluu.oxd.server.model.UmaResource umaResource = site.umaResource(params.getPath(), params.getHttpMethod());
+        if (umaResource == null) {
+            final ErrorResponse error = new ErrorResponse("invalid_request");
+            error.setErrorDescription("Resource is not protected with path: " + params.getPath() + " and httpMethod: " + params.getHttpMethod() +
+                    ". Please protect your resource first with uma_rs_modify command. Check details on " + CoreUtils.DOC_URL);
+            LOG.error(error.getErrorDescription());
+            throw new WebApplicationException(Response
+                    .status(Response.Status.BAD_REQUEST)
+                    .type(MediaType.APPLICATION_JSON_TYPE)
+                    .entity(Jackson2.asJson(error))
+                    .build());
+        }
+
+        UmaMetadata discovery = getDiscoveryService().getUmaDiscoveryByOxdId(params.getOxdId());
+        UmaResourceService resourceService = UmaClientFactory.instance().createResourceService(discovery, getHttpService().getClientExecutor());
+
+        UmaResource opUmaResource = getResource(resourceService, params, umaResource.getId());
+
+        try {
+            String pat = getUmaTokenService().getPat(params.getOxdId()).getToken();
+            resourceService.updateResource("Bearer " + pat, umaResource.getId(), opUmaResource);
+        } catch (ClientResponseFailure e) {
+            LOG.debug("Failed to update resource. Entity: " + e.getResponse().getEntity(String.class) + ", status: " + e.getResponse().getStatus(), e);
+            if (e.getResponse().getStatus() == 400 || e.getResponse().getStatus() == 401) {
+                LOG.debug("Try maybe PAT is lost on AS, force refresh PAT and re-try ...");
+                resourceService.updateResource("Bearer " + getUmaTokenService().obtainPat(params.getOxdId()), umaResource.getId(), opUmaResource); //refresh pat and update
```

Comment:
Here we refreshed PAT and updateResource(). Lets assume that it works successfully. And then on next line we will `throw e` (line 89). Looks wrong, no? We should update RP and return response if refresh of PAT and further update is successful.  Create method where when you update resource you also update RP and return response. In this case you will never create bug that resource is updated on OP but not on db or error is thrown even if update operation was successful.

```
public RsModifyResponse update(String pat, Rp rp, UmaResourceService resourceService, UmaResource resource) {
    resourceService.updateResource("Bearer " + pat, resource.getId(), resource);
    updateRp(opUmaResource, rp, umaResource.getId());

    return new RsModifyResponse(rp.getOxdId());
}
```

Reply:
Yeah, it was a mistake. Now, corrected the error.